### PR TITLE
fix #284235: implement lyricsMinDistance

### PR DESCRIPTION
**Pull request by MarcSabatella**
In 2.3.2, the "Minimum note distance" style setting was the only thing that determined minimum space between lyrics.  For 3.0, we start there, but then also add an addition 1sp if space between lyrics.  This is way too much and causes scores to take more space than before.

This PR implements a new lyricsMinDistance style setting that is added on top of the minimu  note distance.  I have it default to 0 right now, so default spacing is identical to 2.3.2 (except that 2.3.2 did not force hyphens to display whereas 3.0 does by default, which I think will prevent any complaints we are still too tightly spaced).

I expect there may be test failures due to the new style setting, but perhaps not.

Adding the style setting also gave me the opportunity to fix the tab order on the Lyrics page of the style dialog - it was terrible.  Others need attention as well, but another day.

### CODE REVIEW

**Comment by MarcSabatella** [comment]
I'm marking this work in progress as more tweaks may be needed to get good behavior with respect to hyphens.  Also, seems I still don't quite have the tab order right.

**Comment by MarcSabatella** [comment]
Got the hyphen thing worked out.  The force hyphen option actually didn't affect spacing at all, only whether the hyphen would show up at all.  So it depended on ChordRest::shape() to add space, which was happening even when not needed.  I have now updated that to add just enough space for the hyphen, and only when needed.  I tested it with all sorts of different combinations of minimum note distance, minimum lyrics distance, and minimum hyphen length, and all works as I think it should.

As for the tab order, the remaining issue is the same one shared by other pages in the style dialog - the OffsetSelect widgets are not well-behaved here.  So that needs to be addressed more globally.

Bottom line: I think this is good, but I welcome review and testing, of course!

**Comment by MarcSabatella** [comment]
Tests passed with the first commit.  Now added a second to handle some compatibility for read206 & read114, these will probably need updates to compat tests.

**Comment by anatoly-os** [comment]
Rebase is needed.

**Comment by MarcSabatella** [comment]
Rebased using GitHub, but also doing it again locally to get rid of that extra "merge" commit

**Comment by MarcSabatella** [comment]
Good to go.  Should make a noticeable difference in vocal scores, allowing them to be more compact by default but also allowing control over the lyric spacing.

**Comment by Jojo-Schmitz** [comment]
Please add to 3.0.5 too
